tests/foreman/cli/test_discoveredhost.py

```diff
@@ -836,3 +836,63 @@ def test_positive_update_discover_hostname_settings(self):
 
         :CaseImportance: High
         """
+
+    @skip_if_bug_open('bugzilla', 1572947)
+    @tier3
+    @upgrade
+    def test_positive_provision_pxe_host_with_parameters(self):
+        """Provision the pxe-based BIOS discovered host with host parameters from cli using
+        SYSLINUX loader
+
+        :id: 4b315fe1-2eba-4e59-bd0a-9d16ce319ce2
+
+        :Setup:
+            1. Create a BIOS VM and set it to boot from a network
+
+        :steps:
+            1. Build a default PXE template
+            2. Configure HostGroup to be used for provisioning the discovered host
+               after discovery
+            3. PXE Boot the VM (from Network)
+            4. Wait for Host to be discovered by Satellite
+            5. Provision the discovered host with host parameters
+
+        :expectedresults:
+            1. Ensure host appeared in Discovered Hosts on satellite
+            2. Ensure the host is provisioned with correct and all the host parameters
+               given during provisioning
+            3. Ensure the entry from discovered host list disappeared
+
+        :CaseImportance: High
+        """
+        param1_key, param1_value = gen_string('alpha'), gen_string('alphanumeric')
+        param2_key, param2_value = gen_string('alpha'), gen_string('alphanumeric')
+        host_params = [
+            '{0}={1}, {2}={3}'.format(param1_key, param1_value, param2_key, param2_value)]
+        # fixme: assertion #1
+        if not self.configured_env:
+            self.__class__.configured_env = configure_env_for_provision(
+                org=self.org, loc=self.loc)
+        with LibvirtGuest() as pxe_host:
+            hostname = pxe_host.guest_name
+            # fixme: assertion #2-3
+            # assertion #4
+            discovered_host = self._assertdiscoveredhost(hostname)
```

Comment:
_assertdiscoveredhost method name kind wrong, as we are only trying to get discovered host after making multiple iterations. Yes, we can fix this in some other PR.

Reply:
Yes... just reusing this method created 1000 years ago :)